src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx

```diff
@@ -82,8 +84,19 @@ const EditSamplesConfirmationModal = ({
   };
 
   const handleClose = function () {
+    setLoseProgressModalOpen(true);
+  };
+
+  const handleCloseAfterConfirmation = function () {
+    setLoseProgressModalOpen(true);
     clearState();
     onClose();
+    setHasImportedMetadataFile(false);
+    setLoseProgressModalOpen(false);
+  };
+
+  const handleSaveProgressModalClose = function () {
```

Comment:
I'm slightly confused why one is SaveProgressModal and the other is LoseProgressModal if they both refer to the same component